test_stack.py

```diff
@@ -0,0 +1,34 @@
+import io
+import sys
+
+import pytest
+
+from stack import Stack
+
+
+def test_push():
+    s = Stack([7, 3, "yes"])
+    s.push("please")
+
+    assert len(s) == 4
```

Comment:
Is this enough to know that push did the right thing?  Maybe pop also here to ensure that what you pushed on is retrievable.